demo/client/src/components/FilesList.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import config from '../config.json'; // eslint-disable-line import/no-unresolved
+
+const demoServer = `${config.ServiceEndpoint}/demo`;
+
+class FilesList extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      loading: true,
+      files: [],
+      links: {},
+    };
+
+    this.generateLink = this.generateLink.bind(this);
+    this.loadFiles = this.loadFiles.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadFiles();
+  }
+
+  loadFiles() {
+    this.setState({
+      loading: true,
+    });
+
+    window
```

Comment:
Two questions here.
Why not using fetch ([supported](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#supported-language-features-and-polyfills)) directly and relying on the catch?

```js
    fetch(`${demoServer}/meta`)
      .then(res => res.json())
      .then(data => {
        this.setState({
          loading: false,
          files: data,
        });
      })
      .catch(error => {
        console.log(`An error happened: ${error.message}`);
      });
```

Reply:
The `handleErrors` thingy would be nice indeed :)